### PR TITLE
refactor(root): Nv 3846 consolidate env file behavior

### PR DESCRIPTION
**Pull request by denis-kralj-novu**
### What changed? Why was the change needed?
web and widget were the 2 apps that had their `.env` file included in the repository.  api, ws and worker have a procedure in place for how to setup the initial `.env` state via setup script, so this change aims to consolidate the behavior and make it as similar as possible without affecting local development or deployed behavior

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -97,7 +97,7 @@ fabric.properties
 # Module directory
 .terraform/
 
-.env
+**/.env
```
Comment (denis-kralj-novu) [inline .gitignore:100]: All of the removal in other .gitignore files are related to consolidation done here on the top level

File: apps/web/.example.env
Comment (denis-kralj-novu) [inline apps/web/.example.env:1]: shows up like a move since the content is the same

File: apps/widget/.example.env
Comment (denis-kralj-novu) [inline apps/widget/.example.env:1]: shows up like a move since the content is the same

File: scripts/setup-env-files.js [hunk 2]
```diff
@@ -15,14 +16,10 @@ const prePopulateEnv = (folders, folderBasePath, exampleEnvFilePath = 'src/.exam
...
-  const apps = ['api', 'ws', 'worker'];
   const appsBasePath = `${__dirname}/../apps`;
-
   console.log('----------------------------------------');
-  console.log('Pre-populating .env files from .example.env');
-
-  prePopulateEnv(apps, appsBasePath);
-
+  prePopulateEnv(['api', 'ws', 'worker'], appsBasePath);
+  prePopulateEnv(['web', 'widget'], appsBasePath, '.example.env', '.env');
```
Comment (denis-kralj-novu) [inline scripts/setup-env-files.js:22]: web and widget don't have their `.env` files in `/src` but in the parent directory, so this adjustment is needed

File: .gitignore (hunk 1 above)
Comment (rifont) [reply .gitignore:100]: 🧹 nice one!

**Review by rifont — COMMENTED** [review]
Looks great, nice tidy up and move to consistency.

@LetItRock , do you see any issues with this simplification? The main point to verify is that the population to the `window` object with Web Env-vars is completed. Tagging you as you have familiarity with this config step.

File: scripts/setup-env-files.js (hunk 2 above)
Comment (SokratisVidros) [reply scripts/setup-env-files.js:22]: 😮‍💨

**Review by SokratisVidros — APPROVED** [review]

**Comment by denis-kralj-novu** [comment]
@SokratisVidros once this is merged it will deploy to `dev.web.novu.co` correct? Do we have some way to deploy to a temp env to ensure I haven't missed anything in this env shuffling? I don't want to get into a situation where I break dev 😁

**Comment by SokratisVidros** [comment]
@denis-kralj-novu You can use a Netlify preview deployment for Web app changes. We don't have preview deployment infra on AWS yet. So let's merge it but be ready to rollback [see relevant workflow](https://github.com/novuhq/novu/actions/workflows/rollback.yml).

**Comment by denis-kralj-novu** [comment]
> @denis-kralj-novu You can use a Netlify preview deployment for Web app changes. We don't have preview deployment infra on AWS yet. So let's merge it but be ready to rollback [see relevant workflow](https://github.com/novuhq/novu/actions/workflows/rollback.yml).

I will merge first thing tomorrow morning then.

**Comment by LetItRock** [comment]
> Looks great, nice tidy up and move to consistency.
> 
> @LetItRock , do you see any issues with this simplification? The main point to verify is that the population to the `window` object with Web Env-vars is completed. Tagging you as you have familiarity with this config step.

@rifont @denis-kralj-novu 
The `.env` file for the Web application is created during the cloud deployment to prod or dev env: https://github.com/novuhq/novu/blob/e246a7539f8487d1b873597b410a7c50806981c7/.github/workflows/reusable-web-deploy.yml#L75

Then we create the `env-config.js` based on `.env` in the `public` folder which is imported as a script by `index.html`.

A similar process we do for the docker images, but we read from the local environment variables.

---

During the local development, the fallback values are used as the configuration: https://github.com/novuhq/novu/blob/e246a7539f8487d1b873597b410a7c50806981c7/libs/shared-web/src/config.ts#L18
For the future: if we would like to get rid of fallback values we need to make changes in the `config.ts` and fill the `.env` file.

---

But overall, it looks good to me 😄 😛

**Review by LetItRock — APPROVED** [review]

**Pull request merged**